maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveJsonSchema.java

```diff
@@ -89,17 +91,27 @@ protected ObjectNode mergeMultipleDataTypes(ObjectNode mergedArray,
     return mergedArray;
   }
 
-  public ObjectNode getSchemaForMultipleMessages(List<String> messages)
+  /**
+   * Treated same as array of records, the items derived is returned as schema
+   * Exactly one schema is returned
+   */
+  @Override
+  public ObjectNode getSchemaForMultipleMessages(List<JsonNode> messages)
       throws JsonProcessingException {
-    // Get schema for multiple messages. Exactly one schema is returned
-    // Treated same as array of records, the items derived is returned
-    List<JsonNode> messageObjects = new ArrayList<>();
-    for (String message : messages) {
-      messageObjects.add(mapper.readTree(message));
-    }
-    JsonNode schema = getSchemaForArray(messageObjects, "").get("items");
+    JsonNode schema = getSchemaForArray(messages, "").get("items");
+    convertToFormat(schema, "");
     ObjectNode schemaInformation = mapper.createObjectNode();
     schemaInformation.set("schema", schema);
     return schemaInformation;
   }
+
+  /**
+   * Generate json schema and check for any errors
+   */
+  protected JsonNode convertToFormat(JsonNode schema, String name) {
+    JsonSchema jsonSchema = new JsonSchema(schema);
+    jsonSchema.validate();
+    return schema;
```

Comment:
Is the intent to return the input value as the function output without any modification?